pkg/storaged/lvol-tabs.jsx

```diff
@@ -250,19 +267,84 @@
+
+    let initial_pvs;
+
+    if (!has_subvols)
+        initial_pvs = pvs_as_spaces;
+    else {
+        initial_pvs = [];
+
+        // Step 1: Find the spaces that are already used for a
+        // subvolume.  If a subvolume uses more than one, prefer the
+        // one with more available space.
+        for (const sv of subvols) {
+            let sel = null;
+            for (const p in sv) {
+                for (const spc of pvs_as_spaces)
+                    if (spc.block.path == p && (!sel || sel.size < spc.size))
+                        sel = spc;
+            }
+            if (sel)
+                initial_pvs.push(sel);
+        }
+
+        // Step 2: Select missing one randomly.
+        for (const pv of pvs_as_spaces) {
+            if (initial_pvs.indexOf(pv) == -1 && initial_pvs.length < subvols.length)
+                initial_pvs.push(pv);
+        }
+    }
+
+    function max_size(spaces) {
+        const layout = lvol.Layout;
+        const pvs = spaces.map(s => s.pvol);
+        const n_pvs = pvs.length;
+        const sum = pvs.reduce((sum, pv) => sum + pv.FreeSize, 0);
+        const min = Math.min.apply(null, pvs.map(pv => pv.FreeSize));
+
+        if (!has_subvols) {
+            return sum;
+        } else if (layout == "raid0") {
+            return n_pvs * min;
+        } else if (layout == "raid1") {
+            return min;
+        } else if (layout == "raid10") {
+            return (n_pvs / 2) * min;
+        } else if ((layout == "raid4" || layout == "raid5")) {
+            return (n_pvs - 1) * min;
+        } else if (layout == "raid6") {
+            return (n_pvs - 2) * min;
+        } else
+            return 0; // not-covered: internal error
+    }
+
     let grow_size;
-    let size_fields = [];
+    const size_fields = [];
     if (!to_fit) {
-        size_fields = [
+        if (has_subvols || lvol.Layout == "linear")
+            size_fields.push(
+                SelectSpaces("pvs", _("Physical Volumes"),
+                             {
+                                 spaces: pvs_as_spaces,
+                                 value: initial_pvs,
+                                 min_selected: subvols.length,
+                                 validate: val => {
+                                     if (has_subvols && subvols.length != val.length)
+                                         return cockpit.format(_("Exactly $0 physical volumes must be selected"),
+                                                               subvols.length);
+                                 }
+                             }));
+        size_fields.push(
             SizeSlider("size", _("Size"),
                        {
                            value: lvol.Size,
                            min: lvol.Size,
-                           max: (pool ? pool.Size * 3 : lvol.Size + vgroup.FreeSize),
+                           max: (pool ? pool.Size * 3 : lvol.Size + max_size(initial_pvs)),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230912-152256-2d3c74b2-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)